src/sinks/aws_cloudwatch_logs/mod.rs

```diff
@@ -351,29 +352,26 @@ impl Service<Vec<InputLogEvent>> for CloudwatchLogsSvc {
     type Error = CloudwatchError;
     type Future = request::CloudwatchFuture;
 
-    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+    fn poll_ready(&mut self, cx: &mut Context) -> Poll<Result<(), Self::Error>> {
         if let Some(rx) = &mut self.token_rx {
-            match rx.poll() {
-                Ok(Async::Ready(token)) => {
+            match ready!(Pin::new(rx).poll(cx)) {
+                Ok(token) => {
                     self.token = token;
                     self.token_rx = None;
-                    Ok(().into())
                 }
-                Ok(Async::NotReady) => Ok(Async::NotReady),
-                Err(oneshot::Canceled) => {
+                Err(_) => {
```

Comment:
I would still match the concrete error type here tbh, makes it a bit more clear :)

Reply:
I'd happy do this, I tried initially, but looks like this impossible?
```bash
error[E0603]: tuple struct constructor `RecvError` is private
   --> src/sinks/aws_cloudwatch_logs/mod.rs:360:37
    |
360 |                 Err(oneshot::error::RecvError(_)) => {
    |                                     ^^^^^^^^^ private tuple struct constructor
    | 
   ::: /home/kirill/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-0.2.21/src/sync/oneshot.rs:40:26
    |
40  |     pub struct RecvError(pub(super) ());
    |                          ------------- a constructor is private if any of the fields is private
```

Reply:
Hm, you're right....sigh, I don't think there is a better way to fix this.

Reply:
It looks like this is `pub` in tokio 0.2.22: https://docs.rs/tokio/0.2.22/tokio/sync/oneshot/error/struct.RecvError.html

Reply:
`RecvError` is `pub` in 0.2.21 too, but there is not changes for creating error outside of tokio crate.
https://docs.rs/tokio/0.2.21/src/tokio/sync/oneshot.rs.html#40
https://docs.rs/tokio/0.2.22/src/tokio/sync/oneshot.rs.html#40

Reply:
Oooohhhh, I missed that bit in the constructor. Too bad.